### PR TITLE
Framework.md: Prototyped hierarchical TOC and headings

### PR DESCRIPTION
**Pull request by jonathan-r-thorpe**
* Made the TOC hierarchical 
* Made heading level match TOC nesting
* Ordered TOC and document sections alphabetically within each level

### CODE REVIEW

**Comment by cristian-recoseanu** [comment]
Hi Jonathan.

I think for the datatypes having them sorted alphabetically is fine but for classes we should use the inheritance level and index still.
That means NcObject comes first, then NcBlock, then NcWorker and then NcManager.

File: docs/Framework.md [hunk 1]
```diff
@@ -805,120 +818,100 @@ enum NcResetCause {
...
-interface NcMethodResultError: NcMethodResult {
-    attribute NcString    errorMessage; // Error message
+// Touchpoint resource class
+interface NcTouchpointResource {
+    attribute NcString    resourceType; // The type of the resource
 };
 ```
 
-### NcMethodResultPropertyValue
+The `contextNamespace` attribute is inherited from NcTouchpoint and can only be `x-nmos/channelmapping`
```
Comment (cristian-recoseanu) [inline docs/Framework.md]: This needs to be moved under the NcTouchpointNmosChannelMapping section.

File: docs/Framework.md [hunk 2]
```diff
@@ -805,120 +818,100 @@ enum NcResetCause {
...
-    attribute sequence<NcBlockMemberDescriptor>    value; // Block member descriptors method result value
+// Touchpoint resource class for NMOS resources
+interface NcTouchpointResourceNmosChannelMapping: NcTouchpointResourceNmos {
+    attribute NcString    ioId; // IS-08 Audio Channel Mapping input or output id
 };
 ```
 
-### NcMethodResultClassDescriptor
+The `resourceType` attribute is inherited from NcTouchpointResource and can only be: `input, output`.
+
```
Comment (cristian-recoseanu) [inline docs/Framework.md:899]: Remove extra newline.

**Review by cristian-recoseanu — COMMENTED** [review]
I think for the datatypes having them sorted alphabetically is fine but for classes we should use the inheritance level and index still.
That means NcObject comes first, then NcBlock, then NcWorker and then NcManager.

File: docs/Framework.md (hunk 1 above)
Comment (jonathan-r-thorpe) [reply docs/Framework.md]: Fixed in latest commit

File: docs/Framework.md (hunk 2 above)
Comment (jonathan-r-thorpe) [reply docs/Framework.md:899]: Fixed in latest commit

**Comment by garethsb** [comment]
Is any of this auto generated?

**Comment by jonathan-r-thorpe** [comment]
No, this has been authored, but based on @cristian-recoseanu's original auto-generated doc.

The purpose of this pull request was really to prototype an alternative layout which Cristian could then fold into that auto-generation script.

File: docs/Framework.md [hunk 3]
```diff
@@ -224,12 +218,9 @@ NcWorker is the base class for any worker control class in the control model.
 };
 ```
 
-### Managers
-
+### NcManager
 Further explanations are provided in a dedicated [Managers](Managers.md) section.
```
Comment (cristian-recoseanu) [inline docs/Framework.md:223]: We need to have a new line between this heading and the paragraph.

**Review by cristian-recoseanu — APPROVED** [review]

**Pull request merged**